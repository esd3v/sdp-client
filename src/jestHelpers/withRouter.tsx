import * as React from 'react';
import {Router} from 'react-router-dom';
import {createBrowserHistory} from 'history';

const customHistory = createBrowserHistory();

export const withRouter = (Component: React.FunctionComponent) => (
  <Router history={customHistory}>
    <Component/>
  </Router>
);
