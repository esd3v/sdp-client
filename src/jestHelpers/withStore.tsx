import * as React from 'react';
import {Provider} from 'react-redux';
import {store} from 'store';

export const withStore = (Component: React.FunctionComponent) => (
  <Provider store={store}>
      <Component/>
  </Provider>
);
