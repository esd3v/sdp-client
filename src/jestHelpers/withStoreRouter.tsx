import * as React from 'react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createBrowserHistory} from 'history';
import {store} from 'store';

const customHistory = createBrowserHistory();

export const withStoreRouter = (Component: React.FunctionComponent) => (
  <Provider store={store}>
    <Router history={customHistory}>
      <Component/>
    </Router>
  </Provider>
);
