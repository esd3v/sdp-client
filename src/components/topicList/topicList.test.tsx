import {TopicList} from './';
import renderer from 'react-test-renderer';
import {withStore} from 'jestHelpers';

test('renders correctly', () => {
  const tree = renderer
    .create(withStore(TopicList))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
