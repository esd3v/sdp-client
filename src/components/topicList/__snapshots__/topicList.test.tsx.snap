// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 {
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  position: relative;
}

<div
  className="emotion-0"
/>
`;
